### PR TITLE
Andrelo - Topic 4 - DB creation approach

### PR DESCRIPTION
**Pull request by Andrelo9**

### CODE REVIEW

File: MySqlFiles/HighSchool.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS course(
+	course_id int NOT NULL PRIMARY KEY,
+    course_name varchar(255) NOT NULL,
+    assigned_teacher varchar(255),
+	hours_by_week time,   
+    CONSTRAINT assigned_teacher_fk FOREIGN KEY(assigned_teacher)
+    REFERENCES teacher(teacher_id),
+    CONSTRAINT students_fk FOREIGN KEY(assigned_course)
```
Comment (mmassigoge-glb) [inline MySqlFiles/HighSchool.sql]: Check this foreign key, if you are trying to relate students with courses you should first analyze how many students are taking one course and how many courses is taking one student. Then you can model this using the right approach. Let me know the answers to prior questions please

File: MySqlFiles/HighSchool.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
+CREATE DATABASE high_school;
+
+CREATE TABLE IF NOT EXISTS student(
+	registration_number int NOT NULL PRIMARY KEY,
+    dni_number char(8) UNIQUE NOT NULL,
```
Comment (sebastian-garofalo) [inline MySqlFiles/HighSchool.sql]: This could be a number, why use chars?

File: MySqlFiles/HighSchool.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	teacher_id int NOT NULL PRIMARY KEY,
+    first_name varchar(255) NOT NULL,
+    last_name varchar(255) NOT NULL,	
+    birth_date date
+);
+
+CREATE TABLE IF NOT EXISTS course(
+	course_id int NOT NULL PRIMARY KEY,
+    course_name varchar(255) NOT NULL,
+    assigned_teacher varchar(255),
```
Comment (sebastian-garofalo) [inline MySqlFiles/HighSchool.sql]: The assigned teacher should reference the teachers table primary key, that is an int

**Review by sebastian-garofalo — CHANGES_REQUESTED** [review]

File: MySqlFiles/HighSchool.sql [hunk 4]
```diff
@@ -36,23 +38,96 @@ CREATE TABLE IF NOT EXISTS inscript(
...
-    third_note numeric,
-    final_note numeric,
-    registration_number int,
-    course_id int,
+	first_note NUMERIC,
+    second_note NUMERIC,
+    third_note NUMERIC,
+    final_note NUMERIC,
+    registration_number INT,
+    course_id INT,
```
Comment (mmassigoge-glb) [inline MySqlFiles/HighSchool.sql]: This should be a FK also..

File: MySqlFiles/HighSchool.sql [hunk 5]
```diff
@@ -2,32 +2,34 @@
...
+	hours_by_week TIME,   
     CONSTRAINT assigned_teacher_fk FOREIGN KEY(teacher_id)
     REFERENCES teacher(teacher_id)   
 );
 
 CREATE TABLE IF NOT EXISTS inscript(
-	registration_number int,
-    course_id int,
+	registration_number INT,
+    course_id INT,
```
Comment (mmassigoge-glb) [inline MySqlFiles/HighSchool.sql]: this should be a FK also..

**Pull request merged**